lib/grplatt.gi

```diff
@@ -3654,6 +3654,586 @@ function(G)
+        if 720 mod sizeA <> 0 then return 10;
+        else return -1; fi;
+
+    elif series = "L" then
+        d := m;
+        if d = 2 and q = 7 then
+            # if A ~ PGL(2,7) then return 8; fi;
+            # PGL(2,7) is isomorphic to Aut(PSL(2,7)) which is Aut(S) which has a subgroup isomorphic to A
+            # Thus A <~ PGL(2,7); So, the only requirement for isomorphism is that Size(A) = Size(PGL(2,7)) = 336
+            if sizeA = 336 then return 8; fi;
+        elif d > 2 and not (q = 2 and (d = 3 or d = 4)) then
+            #if not A <~ GammaL(d,q) then return 2 * mu(S);
+            if Size(GammaL(d,q)) mod sizeA <> 0 then return 2 * mu(S);
+            else return -1; fi;
+        fi;
+
+    elif series = "2A" # PSU(3,5)
+        and m+1 = 3
+        and q = 5 then
+        #if not A <~ PSigmaU(3,5) then return 126; fi;
+        return -1;
+
+    elif series = "D" then #P\Omega^+
+        d := 2*m;
+        if d = 8 and q = 2
+            and sizeA/sizeS mod 3 = 0
+            then return 3 * mu(S);
+        elif d = 8 and q = 3
+            and sizeA/sizeS mod 3 = 0
+            and sizeA/sizeS mod 12 <> 0
+            then return 3 * mu(S);
+        elif d = 8 and q = 3
+            and sizeA/sizeS mod 12 = 0
+            then return 3360;
+        elif d = 8
+            and sizeA/sizeS mod 3 = 0
+            then return 3 * mu(S);
+        elif q = 3 and d > 7 and
+            sizeA/sizeS mod 3 <> 0 then
+            if Size(PGO(1,d,3)) mod sizeA <>0 # (*)
+            then return (3^(m-1) + 1) * (3^m -1) / 2 ;
+            else return -1; fi;
+        fi;
+
+    elif series = "G" then
+        #if q = 3 and Iso(A,Aut(S)) then return 2 * mu(S);fi;
+        if q = 3 and sizeA = Size(Aut(S)) then return 2 * mu(S); fi;
+        #This is because A is isomorphic to a subgroup in Aut(S) that contains Inn(S) \cong S at any point.
+        b := LogInt(q,3);
+        if 3^b = q and b >1 then
+            #if not A <~ GammaG(q) then return 2 * mu(S); fi;
+            if Size(GF(q))*Size(GaloisGroup(AsField(GF(q),GF(3)))) mod sizeA <> 0 then return 2 * mu(S);
+            else return -1; fi;
+        fi;
+    elif series = "C" #PSp
+        and m = 2
+        and 2^Log2Int(q) = q
+        and q > 3 then
+        #if not A <~ PGammaSp(d,q) then return 2*mu(S); fi;
+        return -1;
+    elif series = "F" and 2^Log2Int(q) = q then
+        #if not A <~ GammaF(d,q) then return 2*mu(S); fi;
+        if sizeS * Size(GaloisGroup(GF(q))) mod sizeA <> 0 then return 2 * mu(S);
+        else return -1; fi;
+    elif series = "E" and m = 6 then
+        #if not A <~ GammaE(d,q) then return 2*mu(S); fi;
+        if sizeS * Size(GaloisGroup(GF(q))) mod sizeA <> 0 then return 2 * mu(S);
+        else return -1; fi;
+    fi;
+    return mu(S);
+end);
+
+##################################################################################
+##
+#F  MinimalFaithfulPermutationDegreeOfSemiSimpleGroup(<G>)
```

Comment:
I know that this might be considered nitpicking, but GAP does not really use `SemiSimple` in this context, but describes such groups as `Trivial Fitting` or `FittingFree`. Since `Semisimple" is already overloaded with many other meanings I would prefer to keep this naming.

Reply:
I have renamed it it `MinimalFaithfulPermutationDegreeOfFittingFreeGroup` now.